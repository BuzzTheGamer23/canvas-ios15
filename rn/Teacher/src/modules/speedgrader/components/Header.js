//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

import React, { Component } from 'react'
import { connect } from 'react-redux'
import {
  View,
  StyleSheet,
  TouchableHighlight,
} from 'react-native'
import { Text } from '../../../common/text'
import i18n from 'format-message'
import type {
  SubmissionDataProps,
} from '../../submissions/list/submission-prop-types'
import SubmissionStatus from '../../submissions/list/SubmissionStatus'
import Avatar from '../../../common/components/Avatar'

export class Header extends Component {
  props: HeaderProps
  state: State

  constructor (props: HeaderProps) {
    super(props)

    this.state = {
      showingPicker: false,
    }
  }

  navigateToContextCard = () => {
    this.props.navigator.show(
      `/courses/${this.props.courseID}/users/${this.props.userID}`,
      { modal: true }
    )
  }

  renderDoneButton () {
    return (
      <View style={styles.doneButton}>
          <TouchableHighlight onPress={this.props.closeModal} underlayColor='white' testID='header.navigation-done'>
            <View style={{ paddingLeft: 20 }}>
              <Text style={{ color: '#008EE2', fontSize: 18, fontWeight: '600' }}>
                {i18n('Done')}
              </Text>
            </View>
          </TouchableHighlight>
        </View>
    )
  }

  renderGroupProfile () {
    const sub = this.props.submissionProps
    let name = this.props.anonymous
      ? (sub.groupID ? i18n('Group') : i18n('Student'))
      : sub.name

    let avatarURL = this.props.anonymous
      ? ''
      : sub.avatarURL

    if (sub.groupID && !this.props.anonymous) {
      return (
        <View style={styles.profileContainer}>
          <View style={{ flex: 1 }}>
            <View style={styles.innerRowContainer}>
              <TouchableHighlight
                onPress={this.showGroup}
                underlayColor='white'
                testID={'header.groupList.button'}>
                  <View style={styles.innerRowContainer}>
                    <View style={styles.avatar}>
                      <Avatar
                        key={sub.userID}
                        avatarURL={avatarURL}
                        userName={name}
                      />
                    </View>
                    <View style={styles.nameContainer}>
                      <Text style={styles.name} accessibilityTraits='header'>{name}</Text>
                      <SubmissionStatus status={sub.status} />
                    </View>
                  </View>
              </TouchableHighlight>
            </View>
          </View>
          {this.renderDoneButton()}
        </View>
      )
    } else {
      return (
        <View style={styles.profileContainer}>
          <View style={styles.avatar}>
            <Avatar
              key={sub.userID}
              avatarURL={avatarURL}
              userName={name}
              onPress={this.navigateToContextCard}
            />
          </View>
          <View style={[styles.nameContainer, { flex: 1 }]}>
            <Text style={styles.name} accessibilityTraits='header'>{name}</Text>
            <SubmissionStatus status={sub.status} />
          </View>
          {this.renderDoneButton()}
        </View>
      )
    }
  }

  render () {
    return (
      <View style={[this.props.style, styles.header]}>
        {this.renderGroupProfile()}
      </View>
    )
  }

  showGroup = () => {
    this.props.navigator.show(
      `/groups/${this.props.submissionProps.groupID}/users`,
      { modal: true },
      { courseID: this.props.courseID }
    )
  }
}

const styles = StyleSheet.create({
  header: {
    backgroundColor: 'white',
  },
  profileContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginTop: 16,
  },
  innerRowContainer: {
    backgroundColor: 'white',
    flexDirection: 'row',
    alignItems: 'center',
  },
  navButtonImage: {
    resizeMode: 'contain',
    tintColor: '#008EE2',
  },
  avatar: {
    width: 40,
    height: 40,
    marginLeft: 16,
  },
  nameContainer: {
    flexDirection: 'column',
    justifyContent: 'space-between',
    marginLeft: 12,
  },
  name: {
    fontSize: 16,
    fontWeight: '600',
  },
  status: {
    fontSize: 14,
  },
  doneButton: {
    backgroundColor: 'white',
    marginRight: 12,
  },
})

export function mapStateToProps (state: AppState, ownProps: RouterProps): HeaderDataProps {
  let assignmentContent = state.entities.assignments[ownProps.assignmentID]
  let quiz = assignmentContent.data.quiz_id && state.entities.quizzes[assignmentContent.data.quiz_id].data
  let course = state.entities.courses[ownProps.courseID]
  let anonymous = assignmentContent.anonymousGradingOn ||
                  quiz && quiz.anonymous_submissions ||
                  course && course.enabledFeatures.includes('anonymous_grading')
  return {
    anonymous,
  }
}

let Connected = connect(mapStateToProps)(Header)
export default (Connected: any)

type RouterProps = {
  courseID: string,
  assignmentID: string,
  userID: string,
  submissionID: ?string,
  submissionProps: SubmissionDataProps,
  closeModal: Function,
  style?: Object,
}

type State = {
  showingPicker: boolean,
}

type HeaderDataProps = {
  anonymous: boolean,
}

type HeaderProps = RouterProps & HeaderDataProps & { navigator: Navigator }
