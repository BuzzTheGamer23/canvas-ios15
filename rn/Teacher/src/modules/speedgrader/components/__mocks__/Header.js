// @flow

export default ('Header': string)
