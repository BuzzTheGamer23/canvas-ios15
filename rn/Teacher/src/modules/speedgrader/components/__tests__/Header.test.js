//
// Copyright (C) 2016-present Instructure, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

// @flow

import _ from 'lodash'
import React from 'react'
import { Header, mapStateToProps } from '../Header'
import renderer from 'react-test-renderer'
import explore from '../../../../../test/helpers/explore'

jest
  .mock('TouchableHighlight', () => 'TouchableHighlight')
  .mock('TouchableOpacity', () => 'TouchableOpacity')
  .mock('../../../../common/components/Avatar', () => 'Avatar')

const templates = {
  ...require('../../../../__templates__/submissions'),
  ...require('../../../../redux/__templates__/app-state'),
  ...require('../../../../__templates__/helm'),
}

let noSubProps = {
  submissionID: null,
  assignmentID: '2',
  courseID: '3',
  userID: '4',
  submissionProps: {
    name: 'Allura',
    avatarURL: 'https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg',
    status: 'none',
    userID: '4',
    grade: 'not_submitted',
    submissionID: null,
    submission: null,
  },
  closeModal: jest.fn(),
  excuseAssignment: jest.fn(),
  gradeSubmission: jest.fn(),
  selectSubmissionFromHistory: jest.fn(),
  selectedIndex: null,
  selectedAttachmentIndex: null,
  anonymous: false,
  navigator: templates.navigator(),
}

let subProps = {
  ...noSubProps,
  submissionID: '1',
  submissionProps: {
    name: 'Allura',
    avatarURL: 'https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg',
    status: 'submitted',
    userID: '4',
    grade: '5',
    submissionID: '1',
    submission: templates.submissionHistory([
      { id: '1', grade: null, submitted_at: '2017-04-26T17:46:00Z' },
      { id: '2', grade: null, submitted_at: '2016-01-01T00:01:00Z' },
    ]),
  },
}

let groupProps = {
  ...subProps,
  submissionProps: {
    ...subProps.submissionProps,
    groupID: '1',
  },
}

describe('SpeedGraderHeader', () => {
  beforeEach(() => jest.resetAllMocks())

  it('renders with no submission', () => {
    let tree = renderer.create(
      <Header {...noSubProps} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders with a submission', () => {
    let tree = renderer.create(
      <Header {...subProps} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders with a group submission', () => {
    let tree = renderer.create(
      <Header {...groupProps} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders with a grade-only submission', () => {
    let props = _.cloneDeep(subProps)
    props.submissionProps.status = 'none'

    let tree = renderer.create(
      <Header {...props} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders with an over-due grade-only submission', () => {
    let props = _.cloneDeep(subProps)
    props.submissionProps.status = 'missing'
    props.submissionProps.submission = null

    let tree = renderer.create(
      <Header {...props} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('anonymizes the avatar and name', () => {
    let tree = renderer.create(<Header {...subProps} anonymous={true} />)
    expect(tree.toJSON()).toMatchSnapshot()
  })

  it('closes the modal', () => {
    let tree = renderer.create(
      <Header {...subProps} />
    ).toJSON()

    const doneButton = explore(tree).selectByID('header.navigation-done') || {}
    doneButton.props.onPress()
    expect(subProps.closeModal).toHaveBeenCalled()
  })

  it('doesnt show the student name when anonymous', () => {
    let tree = renderer.create(
      <Header {...subProps} anonymous={true} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('navigates to the context card when the avatar is pressed', () => {
    let tree = renderer.create(
      <Header {...subProps} />
    ).toJSON()

    let avatar = explore(tree).selectByType('Avatar')
    avatar.props.onPress()
    expect(subProps.navigator.show).toHaveBeenCalledWith(
      `/courses/3/users/4`,
      { modal: true },
    )
  })

  it('doesnt show the group name when anonymous', () => {
    let tree = renderer.create(
      <Header {...groupProps} anonymous={true} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('opens student list when group is tapped', () => {
    let tree = renderer.create(
      <Header {...groupProps} />
    ).toJSON()
    const groupListButton = explore(tree).selectByID(`header.groupList.button`) || {}
    groupListButton.props.onPress()

    expect(groupProps.navigator.show).toHaveBeenCalledWith(
      `/groups/1/users`,
      { modal: true },
      { courseID: '3' },
    )
  })
})

describe('mapStateToProps', () => {
  it('returns the correct data when there is no submission', () => {
    let state = templates.appState({
      entities: {
        assignments: {
          '2': {
            anonymousGradingOn: true,
          },
        },
        submissions: {
          '1': {
            submission: {},
            pending: 0,
            error: null,
            selectedIndex: 3,
          },
        },
      },
    })

    let dataProps = mapStateToProps(state, noSubProps)
    expect(dataProps).toMatchObject({
      anonymous: true,
    })
  })

  it('returns the correct data when there is a submission', () => {
    let state = templates.appState({
      entities: {
        assignments: {
          '2': {
            anonymousGradingOn: true,
          },
        },
        submissions: {
          '1': {
            submission: {},
            pending: 0,
            error: null,
            selectedIndex: 3,
          },
        },
      },
    })

    let dataProps = mapStateToProps(state, subProps)
    expect(dataProps).toMatchObject({
      anonymous: true,
    })
  })
})
