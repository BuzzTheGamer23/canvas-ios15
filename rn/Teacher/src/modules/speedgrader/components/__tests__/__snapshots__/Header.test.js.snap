// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpeedGraderHeader anonymizes the avatar and name 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "white",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginTop": 16,
      }
    }
  >
    <View
      style={
        Object {
          "height": 40,
          "marginLeft": 16,
          "width": 40,
        }
      }
    >
      <Avatar
        avatarURL=""
        onPress={[Function]}
        userName="Student"
      />
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "column",
            "justifyContent": "space-between",
            "marginLeft": 12,
          },
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <Text
        accessibilityTraits="header"
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Object {
              "fontSize": 16,
              "fontWeight": "600",
            },
            Object {
              "fontFamily": ".SFUIDisplay-semibold",
            },
          ]
        }
      >
        Student
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "fontSize": 14,
                "paddingTop": 2,
              },
              undefined,
              Object {
                "color": "#07AF1F",
              },
            ],
            Object {
              "fontFamily": ".SFUIDisplay",
            },
          ]
        }
      >
        Submitted
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "marginRight": 12,
        }
      }
    >
      <TouchableHighlight
        onPress={[Function]}
        testID="header.navigation-done"
        underlayColor="white"
      >
        <View
          style={
            Object {
              "paddingLeft": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
                Object {
                  "color": "#2D3B45",
                  "fontSize": 16,
                },
                Object {
                  "color": "#008EE2",
                  "fontSize": 18,
                  "fontWeight": "600",
                },
                Object {
                  "fontFamily": ".SFUIDisplay-semibold",
                },
              ]
            }
          >
            Done
          </Text>
        </View>
      </TouchableHighlight>
    </View>
  </View>
</View>
`;

exports[`SpeedGraderHeader doesnt show the group name when anonymous 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "white",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginTop": 16,
      }
    }
  >
    <View
      style={
        Object {
          "height": 40,
          "marginLeft": 16,
          "width": 40,
        }
      }
    >
      <Avatar
        avatarURL=""
        onPress={[Function]}
        userName="Group"
      />
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "column",
            "justifyContent": "space-between",
            "marginLeft": 12,
          },
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <Text
        accessibilityTraits="header"
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Object {
              "fontSize": 16,
              "fontWeight": "600",
            },
            Object {
              "fontFamily": ".SFUIDisplay-semibold",
            },
          ]
        }
      >
        Group
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "fontSize": 14,
                "paddingTop": 2,
              },
              undefined,
              Object {
                "color": "#07AF1F",
              },
            ],
            Object {
              "fontFamily": ".SFUIDisplay",
            },
          ]
        }
      >
        Submitted
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "marginRight": 12,
        }
      }
    >
      <TouchableHighlight
        onPress={[Function]}
        testID="header.navigation-done"
        underlayColor="white"
      >
        <View
          style={
            Object {
              "paddingLeft": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
                Object {
                  "color": "#2D3B45",
                  "fontSize": 16,
                },
                Object {
                  "color": "#008EE2",
                  "fontSize": 18,
                  "fontWeight": "600",
                },
                Object {
                  "fontFamily": ".SFUIDisplay-semibold",
                },
              ]
            }
          >
            Done
          </Text>
        </View>
      </TouchableHighlight>
    </View>
  </View>
</View>
`;

exports[`SpeedGraderHeader doesnt show the student name when anonymous 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "white",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginTop": 16,
      }
    }
  >
    <View
      style={
        Object {
          "height": 40,
          "marginLeft": 16,
          "width": 40,
        }
      }
    >
      <Avatar
        avatarURL=""
        onPress={[Function]}
        userName="Student"
      />
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "column",
            "justifyContent": "space-between",
            "marginLeft": 12,
          },
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <Text
        accessibilityTraits="header"
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Object {
              "fontSize": 16,
              "fontWeight": "600",
            },
            Object {
              "fontFamily": ".SFUIDisplay-semibold",
            },
          ]
        }
      >
        Student
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "fontSize": 14,
                "paddingTop": 2,
              },
              undefined,
              Object {
                "color": "#07AF1F",
              },
            ],
            Object {
              "fontFamily": ".SFUIDisplay",
            },
          ]
        }
      >
        Submitted
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "marginRight": 12,
        }
      }
    >
      <TouchableHighlight
        onPress={[Function]}
        testID="header.navigation-done"
        underlayColor="white"
      >
        <View
          style={
            Object {
              "paddingLeft": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
                Object {
                  "color": "#2D3B45",
                  "fontSize": 16,
                },
                Object {
                  "color": "#008EE2",
                  "fontSize": 18,
                  "fontWeight": "600",
                },
                Object {
                  "fontFamily": ".SFUIDisplay-semibold",
                },
              ]
            }
          >
            Done
          </Text>
        </View>
      </TouchableHighlight>
    </View>
  </View>
</View>
`;

exports[`SpeedGraderHeader renders with a grade-only submission 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "white",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginTop": 16,
      }
    }
  >
    <View
      style={
        Object {
          "height": 40,
          "marginLeft": 16,
          "width": 40,
        }
      }
    >
      <Avatar
        avatarURL="https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg"
        onPress={[Function]}
        userName="Allura"
      />
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "column",
            "justifyContent": "space-between",
            "marginLeft": 12,
          },
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <Text
        accessibilityTraits="header"
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Object {
              "fontSize": 16,
              "fontWeight": "600",
            },
            Object {
              "fontFamily": ".SFUIDisplay-semibold",
            },
          ]
        }
      >
        Allura
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "fontSize": 14,
                "paddingTop": 2,
              },
              undefined,
              Object {
                "color": "#8B969E",
              },
            ],
            Object {
              "fontFamily": ".SFUIDisplay",
            },
          ]
        }
      >
        Not Submitted
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "marginRight": 12,
        }
      }
    >
      <TouchableHighlight
        onPress={[Function]}
        testID="header.navigation-done"
        underlayColor="white"
      >
        <View
          style={
            Object {
              "paddingLeft": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
                Object {
                  "color": "#2D3B45",
                  "fontSize": 16,
                },
                Object {
                  "color": "#008EE2",
                  "fontSize": 18,
                  "fontWeight": "600",
                },
                Object {
                  "fontFamily": ".SFUIDisplay-semibold",
                },
              ]
            }
          >
            Done
          </Text>
        </View>
      </TouchableHighlight>
    </View>
  </View>
</View>
`;

exports[`SpeedGraderHeader renders with a group submission 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "white",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginTop": 16,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flexDirection": "row",
          }
        }
      >
        <TouchableHighlight
          onPress={[Function]}
          testID="header.groupList.button"
          underlayColor="white"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "height": 40,
                  "marginLeft": 16,
                  "width": 40,
                }
              }
            >
              <Avatar
                avatarURL="https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg"
                userName="Allura"
              />
            </View>
            <View
              style={
                Object {
                  "flexDirection": "column",
                  "justifyContent": "space-between",
                  "marginLeft": 12,
                }
              }
            >
              <Text
                accessibilityTraits="header"
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontFamily": ".SFUIDisplay",
                    },
                    Object {
                      "color": "#2D3B45",
                      "fontSize": 16,
                    },
                    Object {
                      "fontSize": 16,
                      "fontWeight": "600",
                    },
                    Object {
                      "fontFamily": ".SFUIDisplay-semibold",
                    },
                  ]
                }
              >
                Allura
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontFamily": ".SFUIDisplay",
                    },
                    Object {
                      "color": "#2D3B45",
                      "fontSize": 16,
                    },
                    Array [
                      Object {
                        "fontSize": 14,
                        "paddingTop": 2,
                      },
                      undefined,
                      Object {
                        "color": "#07AF1F",
                      },
                    ],
                    Object {
                      "fontFamily": ".SFUIDisplay",
                    },
                  ]
                }
              >
                Submitted
              </Text>
            </View>
          </View>
        </TouchableHighlight>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "marginRight": 12,
        }
      }
    >
      <TouchableHighlight
        onPress={[Function]}
        testID="header.navigation-done"
        underlayColor="white"
      >
        <View
          style={
            Object {
              "paddingLeft": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
                Object {
                  "color": "#2D3B45",
                  "fontSize": 16,
                },
                Object {
                  "color": "#008EE2",
                  "fontSize": 18,
                  "fontWeight": "600",
                },
                Object {
                  "fontFamily": ".SFUIDisplay-semibold",
                },
              ]
            }
          >
            Done
          </Text>
        </View>
      </TouchableHighlight>
    </View>
  </View>
</View>
`;

exports[`SpeedGraderHeader renders with a submission 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "white",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginTop": 16,
      }
    }
  >
    <View
      style={
        Object {
          "height": 40,
          "marginLeft": 16,
          "width": 40,
        }
      }
    >
      <Avatar
        avatarURL="https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg"
        onPress={[Function]}
        userName="Allura"
      />
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "column",
            "justifyContent": "space-between",
            "marginLeft": 12,
          },
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <Text
        accessibilityTraits="header"
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Object {
              "fontSize": 16,
              "fontWeight": "600",
            },
            Object {
              "fontFamily": ".SFUIDisplay-semibold",
            },
          ]
        }
      >
        Allura
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "fontSize": 14,
                "paddingTop": 2,
              },
              undefined,
              Object {
                "color": "#07AF1F",
              },
            ],
            Object {
              "fontFamily": ".SFUIDisplay",
            },
          ]
        }
      >
        Submitted
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "marginRight": 12,
        }
      }
    >
      <TouchableHighlight
        onPress={[Function]}
        testID="header.navigation-done"
        underlayColor="white"
      >
        <View
          style={
            Object {
              "paddingLeft": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
                Object {
                  "color": "#2D3B45",
                  "fontSize": 16,
                },
                Object {
                  "color": "#008EE2",
                  "fontSize": 18,
                  "fontWeight": "600",
                },
                Object {
                  "fontFamily": ".SFUIDisplay-semibold",
                },
              ]
            }
          >
            Done
          </Text>
        </View>
      </TouchableHighlight>
    </View>
  </View>
</View>
`;

exports[`SpeedGraderHeader renders with an over-due grade-only submission 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "white",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginTop": 16,
      }
    }
  >
    <View
      style={
        Object {
          "height": 40,
          "marginLeft": 16,
          "width": 40,
        }
      }
    >
      <Avatar
        avatarURL="https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg"
        onPress={[Function]}
        userName="Allura"
      />
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "column",
            "justifyContent": "space-between",
            "marginLeft": 12,
          },
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <Text
        accessibilityTraits="header"
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Object {
              "fontSize": 16,
              "fontWeight": "600",
            },
            Object {
              "fontFamily": ".SFUIDisplay-semibold",
            },
          ]
        }
      >
        Allura
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "fontSize": 14,
                "paddingTop": 2,
              },
              undefined,
              Object {
                "color": "#EE0612",
              },
            ],
            Object {
              "fontFamily": ".SFUIDisplay",
            },
          ]
        }
      >
        Missing
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "marginRight": 12,
        }
      }
    >
      <TouchableHighlight
        onPress={[Function]}
        testID="header.navigation-done"
        underlayColor="white"
      >
        <View
          style={
            Object {
              "paddingLeft": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
                Object {
                  "color": "#2D3B45",
                  "fontSize": 16,
                },
                Object {
                  "color": "#008EE2",
                  "fontSize": 18,
                  "fontWeight": "600",
                },
                Object {
                  "fontFamily": ".SFUIDisplay-semibold",
                },
              ]
            }
          >
            Done
          </Text>
        </View>
      </TouchableHighlight>
    </View>
  </View>
</View>
`;

exports[`SpeedGraderHeader renders with no submission 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "white",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginTop": 16,
      }
    }
  >
    <View
      style={
        Object {
          "height": 40,
          "marginLeft": 16,
          "width": 40,
        }
      }
    >
      <Avatar
        avatarURL="https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg"
        onPress={[Function]}
        userName="Allura"
      />
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "column",
            "justifyContent": "space-between",
            "marginLeft": 12,
          },
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <Text
        accessibilityTraits="header"
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Object {
              "fontSize": 16,
              "fontWeight": "600",
            },
            Object {
              "fontFamily": ".SFUIDisplay-semibold",
            },
          ]
        }
      >
        Allura
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "fontSize": 14,
                "paddingTop": 2,
              },
              undefined,
              Object {
                "color": "#8B969E",
              },
            ],
            Object {
              "fontFamily": ".SFUIDisplay",
            },
          ]
        }
      >
        Not Submitted
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "marginRight": 12,
        }
      }
    >
      <TouchableHighlight
        onPress={[Function]}
        testID="header.navigation-done"
        underlayColor="white"
      >
        <View
          style={
            Object {
              "paddingLeft": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
                Object {
                  "color": "#2D3B45",
                  "fontSize": 16,
                },
                Object {
                  "color": "#008EE2",
                  "fontSize": 18,
                  "fontWeight": "600",
                },
                Object {
                  "fontFamily": ".SFUIDisplay-semibold",
                },
              ]
            }
          >
            Done
          </Text>
        </View>
      </TouchableHighlight>
    </View>
  </View>
</View>
`;
